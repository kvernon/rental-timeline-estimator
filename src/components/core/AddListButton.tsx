import styled from '@emotion/styled';
import { IThemeOptions } from '../../theming/IThemeOptions';
import { FontGroups } from '../../theming/fontGroups';
import React from 'react';

const StyledButton = styled.button<{
  theme: IThemeOptions;
}>`
  padding-top: 10px;
  padding-left: 3px;
  padding-right: 3px;
  height: 70px;
  width: 100%;
  font-family: ${(props) => props.theme.typography.get(FontGroups.inputLabel)?.font};
  font-size: ${(props) => props.theme.typography.get(FontGroups.inputLabel)?.size};
  font-weight: ${(props) => props.theme.typography.get(FontGroups.input)?.weight};
  background-color: ${(props) => props.theme.palette.inputBackground};
  border-radius: 0.3rem;
  border-width: 0;
  border-bottom: 5px solid #021c26;
  box-shadow: 0 10px 1px #05465e;
  margin: 0;
  cursor: pointer;
  transition: background-color 0.4s ease-out;

  :hover {
    background-color: ${(props) => props.theme.palette.inputBackgroundFocus};
  }
`;

export interface IAddListButtonProps {
  role: string;
  label: string;
  theme: IThemeOptions;
  onClick: () => void;
}

export function AddListButton(p: IAddListButtonProps) {
  return <StyledButton type="button" role="button" aria-label={p.role} onClick={p.onClick} theme={p.theme} title={p.label} />;
}
