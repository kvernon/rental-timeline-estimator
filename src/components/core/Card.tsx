import styled from '@emotion/styled';
import { IThemeOptions } from '../../theming/IThemeOptions';
import { FontGroups } from '../../theming/fontGroups';

export const Card = styled.div<{ theme: IThemeOptions }>`
  background: ${(props) => `linear-gradient(#7950C5, ${props.theme.palette.panelBackground} 3px)`};
  color: ${(props) => props.theme.typography.get(FontGroups.panelTitle)?.color};
  box-shadow: 0 10px 15px rgba(16, 27, 30, 0.4);
`;
