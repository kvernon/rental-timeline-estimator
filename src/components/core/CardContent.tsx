import styled from '@emotion/styled';

export const CardContent = styled.div`
  padding: 10px;
  margin-top: 0;

  display: block;

  &:last-child {
    padding-bottom: 0;
  }
`;
