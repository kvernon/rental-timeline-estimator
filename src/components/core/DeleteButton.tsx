import styled from '@emotion/styled';
import { TrashAlt } from '@emotion-icons/fa-solid';
import React from 'react';

const DeleteButtonArea = styled.div`
  transition: background 0.4s ease-out;
  width: 45px;
  background: #520606;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  cursor: pointer;

  :hover {
    background: #7e0a0a;
  }
`;

const TrashColored = styled(TrashAlt)`
  color: red;
  width: 50%;
  transform: scale(1, 1.25);
  box-shadow: 0 0 5px #2d0404;
`;

export interface IDeleteProps {
  onClick?: (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
}

export const DeleteButton = (props: IDeleteProps) => {
  return (
    <DeleteButtonArea
      role={'delete-button'}
      onClick={(e) => {
        if (props.onClick) props.onClick(e);
      }}
    >
      <TrashColored role={'trash-can-icon'} />
    </DeleteButtonArea>
  );
};
