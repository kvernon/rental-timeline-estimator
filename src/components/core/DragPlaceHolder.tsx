import styled from '@emotion/styled';

export const DragPlaceholder = styled.div`
  transition: background 0.4s ease-out;
  width: 20px;
  background-color: rgb(56, 56, 56);
  opacity: 1;
  background-image: radial-gradient(ellipse farthest-corner at 4px 4px, rgb(32, 32, 48), rgb(32, 32, 48) 50%, transparent 50%);
  background-size: 4px 4px;
  cursor: pointer;

  :hover {
    background-color: #6e6e6e;
    cursor: move;
  }
`;
