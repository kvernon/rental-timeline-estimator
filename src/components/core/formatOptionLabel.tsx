import { IPropertyDropDownOption } from './IPropertyDropDownOption';
import React, { ReactNode } from 'react';
import { PropertyImage } from './PropertyImage';

export const formatOptionLabel = (data: IPropertyDropDownOption): ReactNode => {
  return <PropertyImage src={data.image} alt={data.label} title={data.label} />;
};
