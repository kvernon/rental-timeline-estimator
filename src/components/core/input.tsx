import styled from '@emotion/styled';

export interface IInputProps {
  hasSpinner?: boolean;
  useUnderlineOnly?: boolean;
}

export const Input = styled.input<IInputProps>`
  flex-grow: 1;
  text-align: right;
  width: 100%;
  height: 59px;
  border-radius: 0.3rem;

  ${(props) =>
    props.useUnderlineOnly === true &&
    `
    transition: border-color 0.4s ease-out;
    border-color: white;
    border-width: 0px 0px 5px 0;
  `}

  ${(props) =>
    props.useUnderlineOnly === false &&
    `
      transition: background-color 0.4s ease-out;
      border-width: 1px;
  `}

  ${(props) =>
    props.hasSpinner === false &&
    `
    &[type=number]::-webkit-inner-spin-button,
    &[type=number]::-webkit-outer-spin-button {
      -webkit-appearance: none;
      margin: 0;
    }
  `}
`;
