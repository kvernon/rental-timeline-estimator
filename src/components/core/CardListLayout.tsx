import { TypographyDiv } from './TypographyDiv';
import { Header5 } from './Header5';
import React from 'react';
import { Stack } from './Stack';
import { useTheme } from '@emotion/react';
import { IThemeOptions } from '../../theming/IThemeOptions';
import styled from '@emotion/styled';
import { Card } from './Card';
import { ICardListLayoutProps } from './ICardListLayoutProps';

const CardPadding = styled(Card)`
  padding: 40px 40px 35px;
`;

export const CardListLayout = (props: ICardListLayoutProps) => {
  const coreTheme = useTheme() as IThemeOptions;
  return (
    <Stack flexGrow={1} direction={'column'}>
      <TypographyDiv>
        <Header5 theme={coreTheme}>{props.title}</Header5>
      </TypographyDiv>
      <CardPadding theme={coreTheme} aria-label={props.title}>
        {props.children}
      </CardPadding>
    </Stack>
  );
};
