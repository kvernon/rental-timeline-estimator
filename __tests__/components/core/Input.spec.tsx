import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import { Input } from '../../../src/components/core/input';

describe('input unit tests', () => {
  describe('and defaults', () => {
    beforeEach(() => {
      render(<Input role="input" />);
    });

    test('should contain styles', () => {
      const entity = screen.getByRole<HTMLInputElement>('input');

      expect(entity).toMatchSnapshot();
    });
  });

  describe('and no spinner', () => {
    beforeEach(() => {
      render(<Input hasSpinner role="input" />);
    });

    test('should contain styles and spinner removal', () => {
      const entity = screen.getByRole<HTMLInputElement>('input');

      expect(entity).toMatchSnapshot();
    });
  });

  describe('and useUnderlineOnly', () => {
    beforeEach(() => {
      render(<Input useUnderlineOnly role="input" />);
    });

    test('should contain styles with no border and underline', () => {
      const entity = screen.getByRole<HTMLInputElement>('input');

      expect(entity).toMatchSnapshot();
    });
  });
});
