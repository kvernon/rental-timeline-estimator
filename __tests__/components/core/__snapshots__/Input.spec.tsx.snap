// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input unit tests and defaults should contain styles 1`] = `
.emotion-0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: right;
  width: 100%;
  height: 59px;
  border-radius: 0.3rem;
}

<input
  class="emotion-0"
  role="input"
/>
`;

exports[`input unit tests and no spinner should contain styles and spinner removal 1`] = `
.emotion-0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: right;
  width: 100%;
  height: 59px;
  border-radius: 0.3rem;
}

<input
  class="emotion-0"
  role="input"
/>
`;

exports[`input unit tests and useUnderlineOnly should contain styles with no border and underline 1`] = `
.emotion-0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: right;
  width: 100%;
  height: 59px;
  border-radius: 0.3rem;
  -webkit-transition: border-color 0.4s ease-out;
  transition: border-color 0.4s ease-out;
  border-color: white;
  border-width: 0px 0px 5px 0;
}

<input
  class="emotion-0"
  role="input"
/>
`;
